lib/teleterm/daemon/daemon.go

```diff
@@ -789,6 +789,18 @@ func (s *Service) AssumeRole(ctx context.Context, req *api.AssumeRoleRequest) er
 		return trace.Wrap(err)
 	}
 
+	for _, gw := range s.gateways {
+		targetURI := gw.TargetURI()
+		if !(targetURI.GetRootClusterURI() == cluster.URI && targetURI.IsKube()) {
```

Comment:
Nit, but I'd check `IsKube` first, given that the majority of users use just a single cluster. 😶‍🌫️